lib/spack/spack/compiler.py

```diff
@@ -261,8 +261,14 @@ def _find_matches_in_path(cls, compiler_names, detect_version, *path):
         prefixes = [''] + cls.prefixes
         suffixes = [''] + cls.suffixes
 
+        def check_cmp_key(check):
+            name = os.path.basename(check[0])
+            idx = compiler_names.index(name)
```

Comment:
This fails when `check[0]` is not in the list, and it causes `spack compiler find` to stall.

Reply:
@tgamblin I think I introduced this bug when I rebased. Fixed it now.